src/rulesets/LinearAlgebra/structured.jl

```diff
@@ -268,3 +268,20 @@ function rrule(::typeof(logdet), X::Union{Diagonal, AbstractTriangular})
     end
     return y, logdet_pullback
 end
+
+#####
+##### Tridiagonal
+#####
+
+function rrule(::Type{Tridiagonal}, dl, d, du)
+    y = Tridiagonal(dl, d, du)
+    @views function ∇Tridiagonal(∂y)
+        return (
+            NoTangent(),
+            diag(∂y[2:end, 1:(end - 1)]),
+            diag(∂y),
+            diag(∂y[1:(end - 1), 2:end]),
```

Comment:
Can't this be
```suggestion
            diag(∂y, 1),
            diag(∂y),
            diag(∂y, -1),
```
Or maybe other order?

Reply:
other order